File: src/main/java/net/ndrei/teslacorelib/gui/BasicRenderedGuiPiece.java
package net.ndrei.teslacorelib.gui;

import net.minecraft.util.ResourceLocation;

/**
 * Created by CF on 2016-12-21.
 */
public class BasicRenderedGuiPiece extends BasicContainerGuiPiece {
    private ResourceLocation texture;
    private int textureX, textureY;

    public BasicRenderedGuiPiece(int left, int top, int width, int height,
                                    ResourceLocation texture, int textureX, int textureY) {
        super(left, top, width, height);

        this.texture = texture;
        this.textureX = textureX;
        this.textureY = textureY;
    }

    @Override
    public void drawBackgroundLayer(BasicTeslaGuiContainer container, int guiX, int guiY, float partialTicks, int mouseX, int mouseY) {
        if (this.texture != null) {
            container.mc.getTextureManager().bindTexture(this.texture);

            container.drawTexturedModalRect(guiX + this.getLeft(), guiY + this.getTop(),
                    this.textureX, this.textureY, this.getWidth(), this.getHeight());
        }
    }
}

File: src/main/kotlin/net/ndrei/teslacorelib/gui/BasicRenderedGuiPiece.kt
package net.ndrei.teslacorelib.gui

import net.minecraft.util.ResourceLocation

/**
 * Created by CF on 2017-06-28.
 */
open class BasicRenderedGuiPiece(left: Int, top: Int, width: Int, height: Int,
                            private val texture: ResourceLocation?, private val textureX: Int, private val textureY: Int)
    : BasicContainerGuiPiece(left, top, width, height) {

    override fun drawBackgroundLayer(container: BasicTeslaGuiContainer<*>, guiX: Int, guiY: Int, partialTicks: Float, mouseX: Int, mouseY: Int) {
        if (this.texture != null) {
            container.mc.textureManager.bindTexture(this.texture)

            container.drawTexturedModalRect(guiX + this.left, guiY + this.top,
                    this.textureX, this.textureY, this.width, this.height)
        }
    }
}
